enhancements/cluster-logging/cluster-logging-kibana-multitenancy.md

```diff
@@ -0,0 +1,188 @@
+---
+title: cluster-logging-kibana-multitenancy
+authors:
+  - "@ewolinetz"
+reviewers:
+  - "@jcantrill"
+  - "@bparees"
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@richm"
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# cluster-logging-kibana-multitenancy
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of updating the Kibana Multitenancy pattern is to no longer rely on the
+[openshift-elasticsearch-plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin/) to
+provide multitenancy for Kibana. In conjunction with the [elasticsearch-proxy](https://github.com/ewolinetz/enhancements/blob/master/enhancements/cluster-logging/cluster-logging-elasticsearch-proxy.md) this would greatly simplify the openshift-elasticsearch-plugin and potentially make it
+obsolete.
+
+## Motivation
+
+Currently the openshift-elasticsearch-plugin intercepts calls from Kibana, rewrites them
+based on the username to match up with an user-specific index, then rewrites the response so that Kibana does not detect that things have been rewritten. This currently requires that we are able to
+keep up with any changes to the Kibana -> Elasticsearch API for intercepting and updating
+the calls, this instead would be handled by Kibana before sending a request to Elasticsearch.
```

Comment:
> It's handed by Kibana because we are making use of the OD plugin?

yes.

> simplified view into the data (e.g. no dynamic population of index-patterns)

Aren't we still doing that at some level in the elasticsearch proxy though?

Reply:
> > simplified view into the data (e.g. no dynamic population of index-patterns)
> 
> Aren't we still doing that at some level in the elasticsearch proxy though?

To confirm what you are saying: The normal usage of OD is to statically seed tenants.  We still need to dynamically seed index-patterns because we have to dynamically create tenants?

Reply:
The normal OD multitenant usage is:
1. user logs in to kibana
2. OD will create a user tenant for that user
3. That user will then be responsible to create their own index patterns, etc.

So we want to still create the roles dynamically for users so that when a user logs in they can be matched with a specific role we create based on their OKD access. If the user is an admin we would go one step further and also specify the tenant pattern `admin` for that user in their role.

Then when they log in the user should see the kibana user tenant (and if an OKD admin, also see the `admin` tenant) listed as available tenants.